composer/utils/object_store/oci_object_store.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""S3-Compatible object store."""
+
+from __future__ import annotations
+
+import os
+import pathlib
+import uuid
+from typing import Callable, Optional, Union
+
+from composer.utils.import_helpers import MissingConditionalImportError
+from composer.utils.object_store.object_store import ObjectStore
+
+__all__ = ['OCIObjectStore']
+
+
+class OCIObjectStore(ObjectStore):
+    """Utility for uploading to and downloading from an OCI bucket.
+
+    Args:
+        bucket (str): The bucket name.
+        prefix (str): A path prefix such as `folder/subfolder/` to prepend to object names. Defaults to ''.
+    """
+
+    def __init__(
+        self,
+        bucket: str,
+        prefix: str = '',
+    ) -> None:
+        try:
+            import oci
+        except ImportError as e:
+            raise MissingConditionalImportError(conda_package='oci',
+                                                extra_deps_group='oci',
+                                                conda_channel='conda-forge') from e
+
+        # Format paths
+        self.bucket = bucket.strip('/')
+        self.prefix = prefix.strip('/')
+        if self.prefix:
```

Comment:
explicitly check empty string (so this doesn't capture all other truthy values

Reply:
oohh right. done